paper/src/main/java/com/github/maxopoly/finale/combat/AsyncPacketHandler.java

```diff
@@ -82,8 +82,9 @@ public void run() {
 
 					if (target == null || target.isDead() || target.isInvulnerable() ||
 							!world.getUID().equals(target.getWorld().getUID()) || !(target instanceof LivingEntity)) {
-						if (entity instanceof CraftVehicle vehicle){
-							vehicle.getHandle().hurt(DamageSource.playerAttack(((CraftPlayer) attacker).getHandle()), (float) ((CraftPlayer) attacker).getHandle().getAttribute(Attributes.ATTACK_DAMAGE).getValue());
+						if (entity instanceof CraftVehicle || entity.getType() == EntityType.ITEM_FRAME){
```

Comment:
I really don't know what changed here but this originally broke in the 1.17/1.18 update, I don't know why we need to check what it actually is because this usually worked with all things from 1.16 and below. 100% on the refactoring list for finale though